src/test/org/streamingpool/core/domain/StreamDependencyTreeTest.java

```diff
@@ -0,0 +1,22 @@
+package org.streamingpool.core.domain;
+
+import org.assertj.core.api.Assertions;
+import org.junit.Test;
+import org.mockito.Mockito;
+import org.streamingpool.core.service.StreamId;
+
+public class StreamDependencyTreeTest {
+
+    @Test
+    public void test() {
+        StreamStreamDependencyTreeImpl graph = new StreamStreamDependencyTreeImpl();
+        StreamId SOURCE = Mockito.mock(StreamId.class);
+        StreamId ANCESTOR_1 = Mockito.mock(StreamId.class);
+        StreamId ANCESTOR_2 = Mockito.mock(StreamId.class);
+        graph.addDependency(SOURCE, ANCESTOR_1);
+        graph.addDependency(SOURCE, ANCESTOR_2);
+
+        Assertions.assertThat(graph.getAncestorsFrom(SOURCE)).containsOnlyOnce(SOURCE, ANCESTOR_1, ANCESTOR_2);
+    }
+
```

Comment:
I don't know how much coverage is actually wanted in this project but I think it would be good to have also one example with 1 source and 2 descendants and actually test that the 2 descendants will not contain each other as ancestors.

Reply:
Test added! As we don't have possibility to test this in bis-streams before releasing it's indeed better to test more than less ; )